cloudygames/views.py

```diff
@@ -1,28 +1,99 @@
 from django.shortcuts import render
+from django.core import serializers
 
-from rest_framework import viewsets, generics, status
+from rest_framework import viewsets, generics, status, filters
 from rest_framework.decorators import api_view
 from rest_framework.response import Response
 
 from cloudygames.serializers import GameSerializer, GameSessionSerializer, PlayerSaveDataSerializer
 from cloudygames.models import Game, GameSession, PlayerSaveData
 
+import django_filters
+import json
+
+
+######################## Filter ############################
+
+
+class GameFilter(django_filters.FilterSet):
+    users = django_filters.CharFilter(name='users__username')
+
+    class Meta:
+        model = Game
+        fields = ['id', 'name', 'publisher', 'users']
+        order_by = ['name']
+        read_only_fields = ('id',)
+
+class GameSessionFilter(django_filters.FilterSet):
+    game = django_filters.CharFilter(name='game__id')
+    player = django_filters.CharFilter(name='player_username')
+
+    class Meta:
+        model = GameSession
+        fields = ['game', 'player']
+
+class PlayerSaveDataFilter(django_filters.FilterSet):
+    game = django_filters.CharFilter(name='game__id')
+    player = django_filters.CharFilter(name='player_username')
+
+    class Meta:
+        model = PlayerSaveData
+        fields = ['game', 'player']
+
+
+######################## ViewSet ############################
+
 class GameViewSet(viewsets.ModelViewSet):
+    queryset = Game.objects.all()
     serializer_class = GameSerializer
+    filter_class = GameFilter
 
     def get_queryset(self):
         is_owned = self.request.query_params.get('owned', 0)
-        if is_owned:
+        if is_owned == '1':
             _user = self.request.user
-            queryset = Game.objects.filter(users=_user)
-        else:
-            queryset = Game.objects.all()
-        return queryset.order_by('name')
+            return Game.objects.filter(users=_user)
+        return Game.objects.all().order_by('name')
 
 class GameSessionViewSet(viewsets.ModelViewSet):
-	serializer_class = GameSessionSerializer
-	queryset = GameSession.objects.all()
+    queryset = GameSession.objects.all()
+    serializer_class = GameSessionSerializer
+    filter_class = GameSessionFilter
+    
+    def get_queryset(self):
+        _user = self.request.user
+        return GameSession.objects.filter(player=_user)
```

Comment:
Consider to return all for staff. You can do something like:

    if self.request.user.is_staff:
        return GameSession.objects.all()
    else:
        return GameSession.objects.filter(player=self.request.user)